uui/settings.ts

```diff
@@ -1,315 +1,1148 @@
-                                        24: '30',
-                                        30: '30',
-                                        36: '36',
-                                        42: '42',
-                                        48: '48', // no design
-                                    },
-                                },
-                            },
-                        },
-                    },
-                    footer: {
-                        switch: {
-                            24: '12',
-                            30: '18',
-                            36: '18',
-                            42: '24',
-                            48: '24',
-                        },
-                        linkButton: {
-                            24: '24',
-                            30: '30',
-                            36: '36',
-                            42: '42',
-                            48: '48',
-                        },
-                    },
-                },
-                mobile: {
-                    header: {
-                        titleSize: '48',
-                    },
-                    footer: {
-                        linkButton: '48',
-                    },
-                    row: '48',
-                    searchInput: '48',
-                },
-                modal: {
-                    row: '36',
-                    padding: '24',
-                },
+                searchInput: '36',
+                row: '36',
+                rowItem: '36',
+                footer: '36',
             },
         },
-        rowAddons: {
-            defaults: {
-                icon: '18',
-                indentUnit: 24,
-                indentWidth: 18,
+    },
+};
+
+interface IconButtonSettings {
+    icons: {
+        dropdownIcon: Icon;
+    }
+}
+
+const iconButtonSettings: IconButtonSettings = {
+    icons: {
+        dropdownIcon: DropdownIcon,
+    },
+};
+
+interface LabeledInputIcons {
+    infoIcon: Icon;
+    fillInfoIcon: Icon;
+}
+
+interface LabeledInputSizes {
+    default: LabeledInputProps['size'];
+}
+
+interface LabeledInputSettings {
+    icons: LabeledInputIcons;
+    sizes: LabeledInputSizes;
+}
+
+const labeledInputSettings: LabeledInputSettings = {
+    icons: {
+        infoIcon: InfoIcon,
+        fillInfoIcon: FillInfoIcon,
+    },
+    sizes: {
+        default: '36',
+    },
+};
+
+interface LinkButtonIcons {
+    dropdownIcon: Icon;
+}
+
+interface LinkButtonSizes {
+    default: LinkButtonProps['size'];
+    defaultWeight: LinkButtonProps['weight'];
+}
+
+interface LinkButtonSettings {
+    icons: LinkButtonIcons;
+    sizes: LinkButtonSizes;
+}
+
+const linkButtonSettings: LinkButtonSettings = {
+    icons: {
+        dropdownIcon: DropdownIcon,
+    },
+    sizes: {
+        default: '36',
+        defaultWeight: 'semibold',
+    },
+};
+
+interface ModalIcons {
+    closeIcon: Icon;
+}
+
+interface ModalSizes {
+    defaultWidth: number;
```

Comment:
move to css-variables
    defaultWidth: number;
    defaultMaxHeight: CSSProperties['height'];
    defaultMobileMaxHeight: CSSProperties['height'];
    defaultHeaderPadding: FlexRowProps['padding'];
    defaultFooterColumnGap: FlexRowProps['columnGap'];
    defaultFooterPadding: FlexRowProps['padding'];
    defaultFooterVPadding: FlexRowProps['vPadding'];